### DISCUSSION TITLE
Codegen as a basis for a new VSCode extension?

### DISCUSSION THREAD

**Discussion by borekb** [dotansimha/graphql-code-generator#4501] [General]
This is probably a wild idea but I ❤️ graphql-code-generator – it's one of the most reliable tools around our GraphQL code. This is in stark contrast to the available VSCode extensions (Prisma's, Apollo's, Kumar's): most of them don't work well (or at all) with monorepos, have weird bugs etc.

I wonder if a new GraphQL VSCode extension could be based on graphql-code-generator – I imagine the codegen running in watch mode in the background, the extension consuming its output and highlight things like querying incorrect fields. The nice thing would be that I'd be configuring all this via a standard `codegen.yml` / `.graphqlrc`, not some "proprietary" format.

Do you think this could work?

**Comment by dotansimha** [comment]
This is a very interesting topic. We were thinking to check the ability to extend TS language service, but didn't have the chance to dive deep into that. 
Codegen can be the basis for many ideas and extensions, one of them is the great package https://github.com/piglovesyou/graphql-let by @piglovesyou . He recently introduced a solution for integration codegen with Babel, so that means IDEs tools can leverage that and even allow inline typings based on GraphQL operations.
